### PR TITLE
Move new role appointment to GDS design system

### PR DESCRIPTION
**Pull request by chriswhitehouse-tw**
## Description

This ports the new role appointments page to the GOV.UK Design System. Then adds the new view in the GOV.UK Design System and conditionally renders the Bootstrap or GOV.UK Design System based on whether the user has the "Preview design system" permission.

#### On Landing
Screenshot 2023-04-03 at 16 33 47

#### Option drop down
Screenshot 2023-04-03 at 16 33 54

#### Error
Screenshot 2023-04-03 at 16 34 06

### Before (with make_current=false)
Screenshot 2023-04-03 at 16 34 18

#### On landing
Screenshot 2023-04-06 at 11 19 30

#### Option drop down
Screenshot 2023-04-06 at 11 19 47

#### Error
Screenshot 2023-04-06 at 11 20 25

#### On landing
Screenshot 2023-04-06 at 11 21 14

## Trello

https://trello.com/c/TxN4gC9r

⚠️ This repo is Continuously Deployed: make sure you [follow the guidance](https://docs.publishing.service.gov.uk/manual/development-pipeline.html#merge-your-own-pull-request) ⚠️

Follow [these steps](https://guides.rubyonrails.org/upgrading_ruby_on_rails.html) if you are doing a Rails upgrade.

### CODE REVIEW

File: app/views/admin/role_appointments/new.html.erb [hunk 1]
```diff
@@ -1,17 +1,36 @@
-<% page_title "Create new appointment to #{@role_appointment.role.name}"  %>
+<% content_for :page_title, "Create new appointment to #{@role_appointment.role.name}"  %>
+<% content_for :title, "Create new appointment to #{@role_appointment.role.name}" %>
+<% content_for :title_margin_bottom, 6 %>
+<% content_for :error_summary, render(Admin::ErrorSummaryComponent.new(object: @role_appointment, parent_class: "role_appointment")) %>
```
Comment (davidgisbey) [inline app/views/admin/role_appointments/new.html.erb:4]: same comment as before about not needing parent_class https://github.com/alphagov/whitehall/pull/7507#discussion_r1156799071
```
<% content_for :error_summary, render(Admin::ErrorSummaryComponent.new(object: @role_appointment) %>
```

File: app/views/admin/role_appointments/new.html.erb [hunk 2]
```diff
@@ -1,17 +1,36 @@
...
+  <div class="govuk-grid-column-two-thirds">
+    <%= form_for [:admin, @role_appointment.role.becomes(Role), @role_appointment] do |form| %>
+      <%= render 'admin/role_appointments/form', form: form %>
 
-    <%= form_for [:admin, @role_appointment.role.becomes(Role), @role_appointment], html: {class: 'well'} do |form| %>
-      <%= render partial: 'admin/role_appointments/legacy_form', locals: { form: form } %>
-
-      <%= form.save_or_cancel cancel: edit_admin_role_path(@role_appointment.role) %>
+      <div class="govuk-button-group">
+        <%= render "govuk_publishing_components/components/button", {
```
Comment (davidgisbey) [inline app/views/admin/role_appointments/new.html.erb:12]: same comment as here https://github.com/alphagov/whitehall/pull/7507#discussion_r1156808780

File: features/step_definitions/role_steps.rb [hunk 3]
```diff
@@ -82,7 +82,13 @@
   click_on using_design_system? ? "Edit#{role.name}" : role.name
   click_on "New appointment"
   select person_name, from: "Person"
-  select_date 1.day.ago.to_s, from: "Started at"
+  if using_design_system?
+    select 1.day.ago.day.to_s, from: "Day"
```
Comment (davidgisbey) [inline features/step_definitions/role_steps.rb:86]: there's a helper method you can use here 

```
  def fill_in_date_and_time_field(date)
    date = Time.zone.parse(date)

    select date.year, from: "Year"
    select date.strftime("%B"), from: "Month"
    select date.day, from: "Day"
    select date.strftime("%H"), from: "Hour"
    select date.strftime("%M"), from: "Minute"
  end
```

so all you'd need to input is 
```
within "#role_appointment_started_at" do
      fill_in_date_and_time_field(1.day.ago.to_s)
    end
```

File: app/views/admin/role_appointments/new.html.erb (hunk 1 above)
Comment (chriswhitehouse-tw) [reply app/views/admin/role_appointments/new.html.erb:4]: ✅

File: app/views/admin/role_appointments/new.html.erb (hunk 2 above)
Comment (chriswhitehouse-tw) [reply app/views/admin/role_appointments/new.html.erb:12]: ✅

File: app/views/admin/role_appointments/new.html.erb [hunk 4]
```diff
@@ -1,17 +1,36 @@
...
-  <div class="col-md-4">
-    <h2>Existing appointments</h2>
-    <%= render partial: "admin/role_appointments/legacy_list", locals: { role_appointments: @role_appointment.role.role_appointments} %>
+  <div class="govuk-grid-column-one-third">
+    <%= render "govuk_publishing_components/components/heading", {
+      text: "Existing appointments",
+      margin_bottom: 4
+    } %>
+
+    <%= render "admin/role_appointments/list", { role_appointments: @role_appointment.role.role_appointments} %>
```
[the remark this answers is not in the document]
Comment (chriswhitehouse-tw) [reply app/views/admin/role_appointments/new.html.erb:34]: ✅

File: features/step_definitions/role_steps.rb (hunk 3 above)
Comment (chriswhitehouse-tw) [reply features/step_definitions/role_steps.rb:86]: Cool, thanks @davidgisbey. I've changed fill_in_date_and_time_field to fill_in_date_fields.

**Comment by davidgisbey** [comment]
Looks good Chris. It looks like the we can clean up of the commit history a bit. 

https://github.com/alphagov/whitehall/pull/7515/commits/e7e303fbe1b807b85e9776ebe3fa9303b5f4aecd can be squashed into your main commit and https://github.com/alphagov/whitehall/pull/7515/commits/d665e76da97f5a40292fb7680e899f82081cf473 looks like it snuck in during a rebase. This commit can just be dropped.

Give me a yell if you need a hand with any of it. Happy to approve other than that. Great work!

**Comment by nnagewad** [comment]
Looking good. Just wondering if the content "Once an appointment been created..." could be [hint text](https://design-system.service.gov.uk/components/select/#select-with-hint).

**Comment by chriswhitehouse-tw** [comment]
Cool, thanks for the review @nnagewad. Code and screenshots now updated. ✅

**Review by davidgisbey — APPROVED** [review]

**Pull request merged**